runtime/j9vm/javanextvmi.cpp

```diff
@@ -684,8 +688,68 @@ JVM_TakeVirtualThreadListToUnblock(JNIEnv *env, jclass ignored)
 JNIEXPORT jobject JNICALL
 JVM_CreateThreadSnapshot(JNIEnv *env, jobject thread)
 {
-	Assert_SC_true(!"JVM_CreateThreadSnapshot unimplemented");
-	return NULL;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions const * const vmfns = vm->internalVMFunctions;
+	j9object_t resultObject = NULL;
+	jobject result = NULL;
+	vmfns->internalEnterVMFromJNI(currentThread);
+
+	if (NULL == thread) {
+		vmfns->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
+	} else {
+		j9object_t threadObject = NULL;
+		BOOLEAN isVirtual = JNI_FALSE;
+		BOOLEAN isAlive = JNI_FALSE;
+		J9VMThread *targetThread = NULL;
+		if (JVMTI_ERROR_NONE != vmfns->getTargetVMThreadHelper(currentThread, thread, JVMTI_ERROR_NONE, 0, &targetThread, &isVirtual, &isAlive)) {
+			goto done;
+		}
+
+		if (!isAlive) {
+			goto done;
+		}
+
+		resultObject = vm->memoryManagerFunctions->J9AllocateObject(currentThread, J9VMJDKINTERNALVMTHREADSNAPSHOT_OR_NULL(vm), J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
```

Comment:
> The new code are not super long in one line. I have no problem viewing it in my editor. The 100 char limit seems too small for a single line.

- But, several lines also exceed 160 characters. Long, horizontally dense call sites (many params + nested calls) are hard to read. Spreading parameters across multiple lines will help improve readability.
- Be cautious with nested function calls. Here, a null class can be passed to `J9AllocateObject`. From my read,
  `J9AllocateObject` will crash if the class is null. Please review other nested call sites for similar issues.